stan/math/gpu/kernels/neg_rect_lower_tri_multiply.hpp

```diff
@@ -102,7 +103,9 @@ const char* neg_rect_lower_tri_multiply_kernel_code = STRINGIFY(
       for (int w = 0; w < WORK_PER_THREAD; w++) {
         const int A_global_col
             = A_global_col_offset + w * THREAD_BLOCK_SIZE_COL;
-        A[A_global_col * A_rows + i + rows + offset] = -acc[w];
+        if (A_global_col < A_rows && (i + rows + offset) < A_rows) {
+          A[A_global_col * A_rows + i + rows + offset] = -acc[w];
```

Comment:
Looks like you found some issues with the code here - are these covered by the new tests? (assuming yes but want to double check).

Reply:
These are covered by the new tests and also by  the lower trinagular inverse tests.

Reply:
I guess my question is, were the lower triangular tests on `develop` wrong? They don't seem to be updated with this PR. Were there failing tests that didn't come up? Or are these only uncovered by new tests that we should maybe add?

Reply:
Oh yeah, I guess the inverse test should test this on their own. Thanks!